torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -316,6 +316,11 @@ def __init__(
         self.num_magnitude_bins = num_magnitude_bins
 
     def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+        # datumbox: This idiom doesn't work here because we do multiple random() calls within the transform.
+        # For example `_get_random_item()` or `torch.rand(())`.
+        # We should revert back to overwriting the `forward()` reverting the majority of refactoring done at
+        # https://github.com/pytorch/vision/pull/6463
+        # this applies to all AutoAugment methods here, not just RandAugment.
```

Comment:
cc @pmeier

Reply:
Ah yes, that was the reason. Good catch!

Reply:
See https://github.com/pytorch/vision/pull/6492/commits/5832c4fa6cff3d041f81b79164c67ce0e202924c

Reply:
Oh it gets messy. Hopefully we can find a more elegant approach to avoid it. Let's not mark this comment as resolved until we have a merged PR that fixes the issue.

Reply:
Addressed by #6492